parsers/lib/validation.py

```diff
@@ -27,6 +32,53 @@ def check_expected_range(datapoint, value, expected_range, logger, key=None):
     return True
 
 
+def validate_production_diffs(
+        datapoints: list, max_diff: dict, logger: logging.Logger):
+    """
+
+    Parameters
+    ----------
+    datapoints: a list of datapoints having a 'production' field
+    max_diff: dict representing the max allowed diff per energy type
+    logger
+
+    Returns
+    -------
+    the same list of datapoints, with the ones having a too big diff removed
+    """
+    # sort datapoins by datetime
+    datapoints = sorted(datapoints, key=lambda x: x['datetime'])
+
+    ok_diff = pd.Series(np.ones_like(datapoints, dtype=bool))
+    for energy, max_diff in max_diff.items():
+        if 'energy' == 'total':
+            series = pd.Series(
+                [sum([nan_to_zero(v)
+                     for en, v in datapoint['production'].items()])
```

Comment:
if you're not using `en` here, you can loop over `datapoint['production'].values()` instead of `items()`.
```suggestion
                     for v in datapoint['production'].values()])
```

Or by numpy.nansum... `[np.nansum(datapoint['production'].items())]` (untested)